### PR TITLE
fix intel deflater on dataproc

### LINKED ISSUE #1784: extraJavaOptions should be provided to driver as well as the executors
extraJavaOptions are provided only to the spark executors by gatk-launch, these same options should probably be set for the driver as well

### PR DESCRIPTION
**Pull request by lbergelson**
fixes #1784  and #1780  which both caused problems with the intel deflater

note: this incorporates the changes in https://github.com/broadinstitute/gatk/pull/1717 because without them it fails with

```
java.lang.UnsupportedOperationException: Directory size listing not supported on GCS.
	at org.broadinstitute.hellbender.utils.gcs.BucketUtils.dirSize(BucketUtils.java:301)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.getRecommendedNumReducers(GATKSparkTool.java:255)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.writeReads(GATKSparkTool.java:237)
	at org.broadinstitute.hellbender.tools.spark.pipelines.PrintReadsSpark.runTool(PrintReadsSpark.java:35)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.runPipeline(GATKSparkTool.java:313)
	at org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram.doWork(SparkCommandLineProgram.java:38)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:102)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:155)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:174)
	at org.broadinstitute.hellbender.Main.instanceMain(Main.java:67)
	at org.broadinstitute.hellbender.Main.main(Main.java:82)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:731)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:181)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:206)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:121)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

example commandline:

```
./gatk-launch PrintReadsSpark -I gs://hellbender/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -O output --apiKey $HELLBENDER_TEST_APIKEY -- --sparkRunner GCS --cluster dataproc-cluster-3 --project broad-dsde-dev
```

### CODE REVIEW

**Comment by lbergelson** [comment]
@akiezun Could you check that this works for you on dataproc and our local cluster?

**Comment by lbergelson** [comment]
I verified that the intel deflater was being loaded in the executors by checking their logs.  Not all executors actually do any work on the cluster I was using, so check one that has output.

File: gatk-launch [hunk 1]
```diff
@@ -23,14 +23,17 @@ BUILD_LOCATION = script +"/build/install/" + projectName + "/bin/"
 GATK_RUN_SCRIPT = BUILD_LOCATION + projectName
 BIN_PATH = script + "/build/libs"
 
+EXTRA_JAVA_OPTIONS="-Dsamjdk.intel_deflater_so_path=libIntelDeflater.so -Dsamjdk.compression_level=1 -DGATK_STACKTRACE_ON_USER_EXCEPTION=true "
```
Comment (akiezun) [inline gatk-launch]: we probably don't want DGATK_STACKTRACE_ON_USER_EXCEPTION on by default, do we? (maybe we do for the alpha period)

**Comment by akiezun** [comment]
This works for me too. I tried 2 scenarios and confirmed igzip by looking at the logs.
- log in onto the cloud VM (master node of the dataproc cluster) and run from there 
- run from my laptop

back to @lbergelson - one question about GATK_STACKTRACE_ON_USER_EXCEPTION - do we want it on by default?

**Comment by lbergelson** [comment]
@akiezun I keep encountering user exceptions on spark that are missing important information.  Until we fix those cases ( #1786 is one example) I think we should have it on for spark.  Seeing a stack trace is less frustrating than seeing "User Exception: null".  I would recommend keeping it on for spark by default until we tested more combinations of inputs and outputs on more machines.   

If you disagree though I can turn it off.

**Comment by akiezun** [comment]
No, I'm fine with that.

**Pull request merged**